dist/tools/esptools/esptool_v3.2.py

```diff
@@ -0,0 +1,4930 @@
+#!/usr/bin/env python3
```

Comment:
Would it make sense to just download [`esptool`](https://github.com/espressif/esptool) on demand like e.g. [bossa](https://github.com/RIOT-OS/RIOT/tree/master/dist/tools/bossa-1.8)?

Reply:
> Would it make sense to just download [esptool](https://github.com/espressif/esptool) on demand like e.g. [bossa](https://github.com/RIOT-OS/RIOT/tree/master/dist/tools/bossa-1.8) or [cc2538-bsl](https://github.com/RIOT-OS/RIOT/tree/master/dist/tools/cc2538-bsl)?

Hm, I'm not sure:
- We can't use the version from [GitHub](https://github.com/espressif/esptool). Instead we have to use the version provided with the ESP-IDF.
- It's just a single file that usually don't change so often.
- We have to the change version to pyhton3.

Although ESP-IDF is pulled as package `pkg/esp32_sdk` during a Make and we could change the Python version by a patch, I don't know if this would be possible with the target `flash-only` even on a system that doesn't have the toolchain installed and on which no full Make is executed. So copying this single file and updating it with the ESP-IDF if necessary seems to me to be the easier way.

Reply:
> We can't use the version from [GitHub](https://github.com/espressif/esptool). Instead we have to use the version provided with the ESP-IDF.

Ah I didn’t know they were different. Might be worth adding a note somewhere.

Reply:
It is documented here: https://github.com/RIOT-OS/RIOT/blob/0eb410d1b755668e09a4057b1dd27851a770d5dc/cpu/esp32/doc.txt#L384